exchange.lisp

```diff
@@ -289,16 +289,14 @@ need-to-use basis, rather than upon initial loading of the exchange API.")
   (:method :around (gate pubkey secret request)
     (rplacd request (call-next-method gate pubkey secret (cdr request)))))
 
-(defmethod perform ((gate gate))
-  (with-slots (input output . #1=(exchange pubkey secret cache)) gate ;¡ has-a !
-    (when (send-blocks-p output) (setf cache (recv input)))
-    (send (slot-value gate 'output) (gate-post . #1#))))
+(defmethod perform recv input ((gate gate))
+  (with-slots (output . #1=(exchange pubkey secret input)) gate ; ¡ has-a !
+    (send output (gate-post . #1#))))
 
 (defun gate-request (gate path &optional options &aux (id (cons path options)))
-  (with-slots (input output) gate
-    (send input (list* id path options))
-    (loop for reply = (recv output) until (eq (car reply) id)
-       do (send output reply) finally (return (values-list (cdr reply))))))
+  (send (slot-channel gate 'input) (list* id path options))
+  (loop for reply = (recv (output gate)) until (eq (car reply) id)
+     do (send (output gate) reply) finally (return (values-list (cdr reply)))))
 
```

Comment:
This code is slightly cleaner for the same reason as before, although the loop is still horrible and must be fixed as an improvement unrelated to the select method combination from chanl.actors.